### PR TITLE
Refactor: Rename Fuel Class to Fuel Instance and Add Unrecognized Field - 850

### LINKED ISSUE #850: Refactor Fuel Type and Category: Rename Fuel Class and Add Unrecognized Field
## Description

This task involves refactoring the existing `Fuel Type` and `Fuel Category` models by renaming the `Fuel Class` table to `Fuel Instance` and adding an `unrecognized` field to the `fuel_type` table. The renamed `Fuel Instance` table will continue to hold custom fields that allow for user-defined values for specific fuel instances, while the new `unrecognized` field will help categorize fuel types.

## Purpose and Benefit to User

The purpose of this refactor is to improve the clarity, organization, and flexibility of fuel tracking within the system. By renaming the `Fuel Class` table to `Fuel Instance` and adding the `unrecognized` field, users will be able to:

1. Work with a more intuitively named table for specific fuel instances.
2. Easily distinguish between recognized and unrecognized fuel types.
3. Access more relevant data for calculations and reporting.

These enhancements will streamline the management of fuel instances, ensure clearer categorization of fuel types, and improve overall data management and user experience.

## Acceptance Criteria

1. Given I am a user, when I interact with fuel instance data, then I see it referred to as "Fuel Instance" instead of "Fuel Class" throughout the system.
2. Given I am a user, when I query for a fuel instance, then the system uses the values from the renamed `Fuel Instance` table (formerly `Fuel Class`).
3. Given I am an admin, when I view or manage fuel instances, then I can see all values for each instance in the renamed table.
4. Given I am a user, when I view the main fuel types list, then I only see recognized fuel types.
5. Given I am a user, when I view the 'other' fuel types list, then I see unrecognized fuel types.

## Development Checklist

1. Rename the `Fuel Class` table to `Fuel Instance` in the database schema.
2. Update all references in the codebase from `Fuel Class` to `Fuel Instance`.
3. Add a new boolean field `unrecognized` to the `fuel_type` table.
4. Seed the `unrecognized` field with appropriate values for existing records.
5. Update all relevant fuel queries to:
   a. Use the renamed `Fuel Instance` table.
   b. Filter fuel types based on the `unrecognized` status for different list views.
6. Ensure all existing relationships and foreign keys are updated to reflect the renamed `Fuel Instance` table.
7. Write migration scripts to update the database schema with these changes.
8. Update filters to display only recognized fuel types in the main fuel types list.
9. Update filters to display unrecognized fuel types in the 'other' fuel types list.
10. Update any UI elements, labels, and documentation to reflect the new "Fuel Instance" terminology.

## Notes

- This refactor aims to improve terminology and fuel type categorization without fundamentally changing the existing data structure.
- The current schema and relationships within the `fuel_type` table should be maintained while implementing these changes.
- Existing custom fields in the `Fuel Class` table (now `Fuel Instance`) such as `custom_carbon_intensity`, `custom_energy_density`, and `custom_units` should remain unchanged.

### PR DESCRIPTION
**Pull request by hamed-valiollahi**
This PR refactors the Fuel Class table to Fuel Instance and adds an unrecognized field to fuel types.

Closes #850

### CODE REVIEW

File: frontend/src/views/AllocationAgreements/_schema.jsx [hunk 1]
```diff
@@ -211,13 +212,12 @@ export const allocationAgreementColDefs = (optionsData, errors) => [
...
     cellStyle: (params) => {
       const style = cellErrorStyle(params, errors)
-      const conditionalStyle =
-        params.data.fuelType === 'Other'
+      const conditionalStyle = params.data.unrecognized
           ? { backgroundColor: '#fff', borderColor: 'unset' }
           : { backgroundColor: '#f2f2f2' }
       return { ...style, ...conditionalStyle }
     },
-    editable: (params) => params.data.fuelType === 'Other',
```
Comment (AlexZorkin) [inline frontend/src/views/AllocationAgreements/_schema.jsx:220]: We have to leave this: params.data.fuelType === 'Other' 
Reason is that we need to allow input of new custom fuel_types if Other is selected.

File: frontend/src/views/FuelSupplies/_schema.jsx [hunk 2]
```diff
@@ -114,12 +115,12 @@ export const fuelSupplyColDefs = (optionsData, errors) => [
     headerName: i18n.t('fuelSupply:fuelSupplyColLabels.fuelTypeOther'),
     cellStyle: (params) => {
       const style = cellErrorStyle(params, errors)
-      const conditionalStyle = /other/i.test(params.data.fuelType)
```
Comment (AlexZorkin) [inline frontend/src/views/FuelSupplies/_schema.jsx:117]: again need to leave the existing logic in place.

**Review by AlexZorkin — CHANGES_REQUESTED** [review]
Looks good, just a few logic pieces we need to keep in place.

**Comment by hamed-valiollahi** [comment]
> Looks good, just a few logic pieces we need to keep in place.

Thanks, Alex, for your feedback! I’ve made the changes.

**Review by AlexZorkin — APPROVED** [review]

**Pull request merged**